docs/architecture/adr-007.md

```diff
@@ -10,88 +13,104 @@ The current way to define those events is not flexible: events are hardcoded, pa
+  - The Chain ID naming standard is expected to be defined on another ADR.
+- Bridge Fee argument is removed because its format is not well defined. It should be encoded in the newly-introduced `extraData` argument when ready.
 
-List of possible action (as of today)
+### Bridge Destination Restriction
 
-- Actions : `CronosCancelSendToChain`
-- Actions after Payment: `CronosSendToAccount`, `CronosSendToIbc`, `CronosSendCroToIbc`, `CronosSendToChain`
+- The new event formats add support to arbitrary destination chains according to the capability of the chain.
+- It is up to the contract to define restriction on the destination, if needed.
 
- 
-With this generic event format, it is possible to add and remove the parameters for those action in the future or even add new actions while "upgrading" the chain. Smart contract won't be force to migrate to a new contract to benefit those functionnalities.
+### Extra Arguments Guidelines
+
+This is a guideline to future bridge features on leveraging the `bytes extraData` arguement:
+
+1. The first 2 bytes are reserved be a version number, to help the chain logic to recognize the format of elements in `extraData` array.
 
+### Other Events
 
+For other bridge events, we recommend to add the `bytes extraData` to them as well.
+
+Possible events affected (as of writing):
+- `__CronosSendToAccount`
+- `__CronosSendToChainResponse`
+- `__CronosCancelSendToChain`
+- `__CronosSendCroToIbc`
+ 
 ## Status
 
 Proposed
 
 ## Consequences
 
 ### Positive
-- A more flexible way to trigger action after executing a smart contract.
-- Possibility to remove or add parameters to existing actions (if we upgrade ibc-go or gravity module for example)
-- Possibility to add new actions and make them available to smart contract developers without redeploying contract
+
+- Add extensibility to contract for future bridge features without introducing new token standard;
+- Possibility to remove or add parameters to existing actions (if we upgrade ibc-go or gravity module for example);
+- Reduce number of upgrade or migrations requirement to existing token contracts.
 
 ### Negative
-- Might have some security concerns if those actions are not correctly "bounded" by a defined scope (smart contract can only move its own asset, action can only be performed from sender account)
-- Not possible to restrict bundled features for smart contract developers once they implement the generic events
 
+- Might have some security concerns if those arguments are not correctly "bounded" by a defined scope (smart contract can only move its own asset, action can only be performed from sender account);
+- Extra data defined in byte array will make the contract less readable.
+- Manpulation of argument inside a dynamic-sized byte array (bytes) is not trivial;
+- Extra complexity for smart contract to restrict future bridge features because the future arguemnt formats will not be known at deploy time.
```

Comment:
```suggestion
- Extra complexity for smart contract to restrict future bridge features because the future argument formats will not be known at deploy time.
```